src/zui/ZUIButton/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { Button, CircularProgress } from '@mui/material';
+import { FC, KeyboardEventHandler, MouseEventHandler } from 'react';
+
+type ZUIButtonType =
+  | 'primary'
+  | 'secondary'
+  | 'tertiary'
+  | 'destructive'
+  | 'warning'
+  | 'loading';
+
+interface ZUIButtonProps {
+  buttonType?: 'button' | 'reset' | 'submit';
```

Comment:
Why is `buttonType` a better name than just `type`? The thing I don't love about is that it uses a prefix ("button") that is already obvious from the context (`ZUIButton`) and not use for any of the other props (e.g `label` which could have been `buttonLabel`.

If the purpose is only to distinguish this from the other `type` property (which I think should be renamed anyway) then maybe calling this `actionType` or something might make sense?

Reply:
i think that i thought about it as sort of "the type on the html button element", but i agree that it is not translated well and does not give meaning to the user of the component! agree to change it along with changing the name of the other "type"-property!